vscode-lean4/src/utils/projectInfo.ts

```diff
@@ -99,63 +105,44 @@ export async function findLeanPackageVersionInfo(uri: Uri) : Promise<[Uri | null
 
 // Find the 'leanpkg.toml' or 'lean-toolchain' in the given package root and
 // extract the Lean version info from it.
-export async function readLeanVersion(packageUri: Uri) : Promise<string | null> {
+export async function readLeanVersion(packageUri: Uri) : Promise<string> {
     const toolchainFileName = 'lean-toolchain';
     const tomlFileName = 'leanpkg.toml';
     if (packageUri.scheme === 'file') {
         const leanToolchain = Uri.joinPath(packageUri, toolchainFileName);
-        if (fs.existsSync(new URL(leanToolchain.toString()))) {
+        if (await fileExists(new URL(leanToolchain.toString()))) {
             return await readLeanVersionFile(leanToolchain);
         }
```

Comment:
```suggestion
        try {
            return await readLeanVersionFile(leanToolchain);
        } catch (e) {} // file does not exist
```
(assuming that `readLeanVersionFile` throws an exception if it can't open the file)

It's a bad idea to check for the existence of a file before reading it, because it introduces a race condition: the file could be deleted between the existence check and reading it.  So it's better to just handle the exception from the read operation.